addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/rekognition/listeners/AsyncFaceResultListener.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * (C) Copyright 2006-2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ * Contributors:
+ *     anechaev
+ */
+package org.nuxeo.ai.rekognition.listeners;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Map;
+
+import org.nuxeo.ai.enrichment.DetectFacesEnrichmentService;
+import org.nuxeo.ai.enrichment.EnrichmentMetadata;
+import org.nuxeo.ai.pipes.types.BlobTextFromDocument;
+import org.nuxeo.ai.rekognition.RekognitionService;
+import org.nuxeo.runtime.api.Framework;
+
+import com.amazonaws.services.rekognition.model.GetFaceDetectionRequest;
+import com.amazonaws.services.rekognition.model.GetFaceDetectionResult;
+
+public class AsyncFaceResultListener extends BaseAsyncResultListener {
+
+    public static final String SUCCESS_EVENT = "asyncRekognitionFaceSuccess";
+
+    public static final String FAILURE_EVENT = "asyncRekognitionFaceFailure";
+
+    @Override
+    protected String getSuccessEventName() {
+        return SUCCESS_EVENT;
+    }
+
+    @Override
+    protected String getFailureEventName() {
+        return FAILURE_EVENT;
+    }
+
+    @Override
+    protected Collection<EnrichmentMetadata> getEnrichmentMetadata(String jobId, Map<String, Serializable> params) {
+        GetFaceDetectionRequest request = new GetFaceDetectionRequest()
+                .withJobId(jobId);
+
+        RekognitionService rs = Framework.getService(RekognitionService.class);
+        GetFaceDetectionResult result = rs.getClient().getFaceDetection(request);
+        DetectFacesEnrichmentService dfes = Framework.getService(DetectFacesEnrichmentService.class);
```

Comment:
I am surprised this works.  `xxxEnrichmentServices` are not Nuxeo components.  They are registered with the `AIComponent` and can be access via the `getEnrichmentService` method.  But that method takes the String service name rather than the class name. eg. `aiComponent.getEnrichmentService("aws.faceDetection");`

Reply:
calling something `..Service` and not making it one is misleading and not compatible with Nuxeo idioms

Reply:
Service is a term that existed a long time before Nuxeo.

Reply:
true. although in context of Nuxeo `Service` is well defined term. Naming something `Service` might confuse developer who will try to call `Framework.getService` and get nothing

Reply:
Discussed removing the `Service` suffix.